<h2>Inciso a <h2>

In [3]:
import pandas as pd
from wget import download
from os import path, listdir

In [4]:
#descargamos si no existen
if not path.exists("ObesityDataSet_raw_and_data_sinthetic.txt"):
    download("https://ignaciorlando.github.io/datasets/data-science/ObesityDataSet_raw_and_data_sinthetic.txt")
else:
    print("No vamos a bajar el archivo porque ya existe")

if not path.exists("ObesityDataSet_raw_and_data_sinthetic.csv"):
    download("https://ignaciorlando.github.io/datasets/data-science/ObesityDataSet_raw_and_data_sinthetic.csv")
else:
    print("No vamos a bajar el archivo porque ya existe")

No vamos a bajar el archivo porque ya existe
No vamos a bajar el archivo porque ya existe


In [5]:
#leemos e imprimos el archovo txt
with open("ObesityDataSet_raw_and_data_sinthetic.txt", "r", encoding="utf-8") as a:
    archivoTXT = a.read()

print(archivoTXT)

Obesidad

La obesidad, que causa problemas físicos y mentales, es un problema de salud global con graves consecuencias. La prevalencia de la obesidad aumenta constantemente, por lo que se necesita nueva investigación que examine los factores que influyen en la obesidad y cómo predecir la aparición de la condición según estos factores.

Los datos fueron recolectados como parte de este trabajo científico: https://www.sciencedirect.com/science/article/pii/S2352340919306985#sec1

Información del Conjunto de Datos

Este conjunto de datos incluye información para la estimación de los niveles de obesidad en individuos de los países de México, Perú y Colombia, en función de sus hábitos alimenticios y condición física. Los datos contienen 17 atributos y 2111 registros, los registros están etiquetados con la variable de clase NObesidad (Nivel de Obesidad), que permite la clasificación de los datos utilizando los valores de Peso Insuficiente, Peso Normal, Sobrepeso Nivel I, Sobrepeso Nivel II, Ob

In [6]:
#leemos e imprimimos el archivo csv
archivoCSV = pd.read_csv("ObesityDataSet_raw_and_data_sinthetic.csv")
print(archivoCSV)

            Age  Gender    Height      Weight        CALC FAVC  FCVC  NCP  \
0     21.000000  Female  1.620000   64.000000          no   no   2.0  3.0   
1     21.000000  Female  1.520000   56.000000   Sometimes   no   3.0  3.0   
2     23.000000    Male  1.800000   77.000000  Frequently   no   2.0  3.0   
3     27.000000    Male  1.800000   87.000000  Frequently   no   3.0  3.0   
4     22.000000    Male  1.780000   89.800000   Sometimes   no   2.0  1.0   
...         ...     ...       ...         ...         ...  ...   ...  ...   
2106  20.976842  Female  1.710730  131.408528   Sometimes  yes   3.0  3.0   
2107  21.982942  Female  1.748584  133.742943   Sometimes  yes   3.0  3.0   
2108  22.524036  Female  1.752206  133.689352   Sometimes  yes   3.0  3.0   
2109  24.361936  Female  1.739450  133.346641   Sometimes  yes   3.0  3.0   
2110  23.664709  Female  1.738836  133.472641   Sometimes  yes   3.0  3.0   

      SCC SMOKE      CH2O family_history_with_overweight       FAF       TU

In [7]:
var = 'FCVC'
categorias = [1, 2, 3]
#analisis
enteros = (archivoCSV[var] % 1 == 0).sum()
no_enteros = (archivoCSV[var] % 1 != 0).sum()

print("Enteros:", enteros)
print("No enteros:", no_enteros)
print(archivoCSV[var].value_counts())

Enteros: 1285
No enteros: 826
FCVC
3.000000    652
2.000000    600
1.000000     33
2.392665      2
2.714447      2
           ... 
2.919526      1
2.724121      1
2.801992      1
2.748971      1
2.680375      1
Name: count, Length: 810, dtype: int64


In [5]:
#imprimimos cabecera del archivo csv
archivoCSV.info()

<class 'pandas.DataFrame'>
RangeIndex: 2111 entries, 0 to 2110
Data columns (total 17 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Age                             2111 non-null   float64
 1   Gender                          2111 non-null   str    
 2   Height                          2111 non-null   float64
 3   Weight                          2111 non-null   float64
 4   CALC                            2111 non-null   str    
 5   FAVC                            2111 non-null   str    
 6   FCVC                            2111 non-null   float64
 7   NCP                             2111 non-null   float64
 8   SCC                             2111 non-null   str    
 9   SMOKE                           2111 non-null   str    
 10  CH2O                            2111 non-null   float64
 11  family_history_with_overweight  2111 non-null   str    
 12  FAF                             2111 non-null

In [6]:
#imprimimos primeros 5 registros
archivoCSV.head()

,Age,Gender,Height,Weight,CALC,FAVC,FCVC,NCP,SCC,SMOKE,CH2O,family_history_with_overweight,FAF,TUE,CAEC,MTRANS,NObeyesdad
0,21.0,Female,1.62,64.0,no,no,2.0,3.0,no,no,2.0,yes,0.0,1.0,Sometimes,Public_Transportation,Normal_Weight
1,21.0,Female,1.52,56.0,Sometimes,no,3.0,3.0,yes,yes,3.0,yes,3.0,0.0,Sometimes,Public_Transportation,Normal_Weight
2,23.0,Male,1.80,77.0,Frequently,no,2.0,3.0,no,no,2.0,yes,2.0,1.0,Sometimes,Public_Transportation,Normal_Weight
3,27.0,Male,1.80,87.0,Frequently,no,3.0,3.0,no,no,2.0,no,2.0,0.0,Sometimes,Walking,Overweight_Level_I
4,22.0,Male,1.78,89.8,Sometimes,no,2.0,1.0,no,no,2.0,no,0.0,0.0,Sometimes,Public_Transportation,Overweight_Level_II


- count: el número de registros con esa columna.
- mean: promedio de valores de esa columna.
- std: desvío estándar.
- min: valor mínimo de la columna.
- 25%: el cuartil 1 de la columna (ordenando los valores de menor a mayor, el que está ubicado al 25% de ese orden).
- 50%: el cuartil 2 de la columna (ordenando los valores de menor a mayor, el que está ubicado al 50% de ese orden - la mediana!)
- 75%: el cuartil 3 de la columna (ordenando los valores de menor a mayor, el que está ubicado al 75% de ese orden)

In [31]:
#imprimimos resumen estadistico
archivoCSV.describe()

,Age,Height,Weight,FCVC,NCP,CH2O,FAF,TUE
count,2111.000000,2111.000000,2111.000000,2111.000000,2111.000000,2111.000000,2111.000000,2111.000000
mean,24.312600,1.701677,86.586058,2.419043,2.685628,2.008011,1.010298,0.657866
std,6.345968,0.093305,26.191172,0.533927,0.778039,0.612953,0.850592,0.608927
min,14.000000,1.450000,39.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,19.947192,1.630000,65.473343,2.000000,2.658738,1.584812,0.124505,0.000000
50%,22.777890,1.700499,83.000000,2.385502,3.000000,2.000000,1.000000,0.625350
75%,26.000000,1.768464,107.430682,3.000000,3.000000,2.477420,1.666678,1.000000
max,61.000000,1.980000,173.000000,3.000000,4.000000,3.000000,3.000000,2.000000


<h2>Inciso b<h2>

<p>
    Analizar los valores de las columnas. ¿Qué tipo de datos hay en cada una? ¿Qué representan? Incluir información del análisis en la notebook.
<p>

**Tenemos como informacion**

```RangeIndex: 2111 entries, 0 to 2110
Data columns (total 17 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Age                             2111 non-null   float64
 1   Gender                          2111 non-null   str    
 2   Height                          2111 non-null   float64
 3   Weight                          2111 non-null   float64
 4   CALC                            2111 non-null   str    
 5   FAVC                            2111 non-null   str    
 6   FCVC                            2111 non-null   float64
 7   NCP                             2111 non-null   float64
 8   SCC                             2111 non-null   str    
 9   SMOKE                           2111 non-null   str    
 10  CH2O                            2111 non-null   float64
 11  family_history_with_overweight  2111 non-null   str    
 12  FAF                             2111 non-null   float64
 13  TUE                             2111 non-null   float64
 14  CAEC                            2111 non-null   str    
 15  MTRANS                          2111 non-null   str    
 16  NObeyesdad                      2111 non-null   str    
dtypes: float64(8), str(9)
memory usage: 280.5 KB
```

**Que representan?**

- $\color{orange}{\text{age}}$: Edad.

- $\color{orange}{\text{Height}}$: Altura.
- $\color{orange}{\text{Weight}}$: Peso.
- $\color{orange}{\text{family\_history\_with\_overweight}}$: Tiene algun familiar que sufre o sufrio sobrepeso? - `Si o no`.
- $\color{orange}{\text{FAVC}}$: Frecuentemente consume comidas con altas calorias? - `Si o no`.
- $\color{orange}{\text{FCVC}}$: Consume frecuentemente vegetales? - `Nunca, a veces, siempre`.
- $\color{orange}{\text{NCP}}$: Numero de comidas principales?
- $\color{orange}{\text{CAEC}}$: Consumo de alimentos entre comidas - `No, a veces, frecuentemente, siempre`.
- $\color{orange}{\text{SMOKE}}$: Fuma? - `Si o no`.
- $\color{orange}{\text{CH2O}}$: Consumeo de agua diario - `Menos que un litro, entre 1 y 2 litros, mas de 2 litros`.
- $\color{orange}{\text{SCC}}$: Controla el consumo de calorias? - `Si o no`.
- $\color{orange}{\text{FAF}}$: Frecuencia de actividad fisica - `0, 1 a 2, 2 a 4, 4 a 5`.
- $\color{orange}{\text{TUE}}$: Tiempo usando dispositivos tecnologicos? - `0 a 2, 3 a 5, mayor a 5`
- $\color{orange}{\text{CALC}}$: Consumo de alcohol? - `no, a veces, frecuentemente, siempre`.
- $\color{orange}{\text{MTRANS}}$: Metodo de transporte usado - `auto, moto, bici, transporte publico, caminata`.
- $\color{orange}{\text{NObeyesdad}}$: Tipo de obesidad - `peso insuficiente, peso normal, sobrepeso nivel i, sobrepeso nivel 2, obsesidad tipo i, obesidad tipo ii, obesidad tipo iii`.
- $\color{orange}{\text{BMI}}$: índice de masa corporal.

**Analisis**
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Age</th>
      <th>Height</th>
      <th>Weight</th>
      <th>FCVC</th>
      <th>NCP</th>
      <th>CH2O</th>
      <th>FAF</th>
      <th>TUE</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>count</th>
      <td>2111.000000</td>
      <td>2111.000000</td>
      <td>2111.000000</td>
      <td>2111.000000</td>
      <td>2111.000000</td>
      <td>2111.000000</td>
      <td>2111.000000</td>
      <td>2111.000000</td>
    </tr>
    <tr>
      <th>mean</th>
      <td>24.312600</td>
      <td>1.701677</td>
      <td>86.586058</td>
      <td>2.419043</td>
      <td>2.685628</td>
      <td>2.008011</td>
      <td>1.010298</td>
      <td>0.657866</td>
    </tr>
    <tr>
      <th>std</th>
      <td>6.345968</td>
      <td>0.093305</td>
      <td>26.191172</td>
      <td>0.533927</td>
      <td>0.778039</td>
      <td>0.612953</td>
      <td>0.850592</td>
      <td>0.608927</td>
    </tr>
    <tr>
      <th>min</th>
      <td>14.000000</td>
      <td>1.450000</td>
      <td>39.000000</td>
      <td>1.000000</td>
      <td>1.000000</td>
      <td>1.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>25%</th>
      <td>19.947192</td>
      <td>1.630000</td>
      <td>65.473343</td>
      <td>2.000000</td>
      <td>2.658738</td>
      <td>1.584812</td>
      <td>0.124505</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>50%</th>
      <td>22.777890</td>
      <td>1.700499</td>
      <td>83.000000</td>
      <td>2.385502</td>
      <td>3.000000</td>
      <td>2.000000</td>
      <td>1.000000</td>
      <td>0.625350</td>
    </tr>
    <tr>
      <th>75%</th>
      <td>26.000000</td>
      <td>1.768464</td>
      <td>107.430682</td>
      <td>3.000000</td>
      <td>3.000000</td>
      <td>2.477420</td>
      <td>1.666678</td>
      <td>1.000000</td>
    </tr>
    <tr>
      <th>max</th>
      <td>61.000000</td>
      <td>1.980000</td>
      <td>173.000000</td>
      <td>3.000000</td>
      <td>4.000000</td>
      <td>3.000000</td>
      <td>3.000000</td>
      <td>2.000000</td>
    </tr>
  </tbody>
</table>
</div>

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: left;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Age</th>
      <th>Gender</th>
      <th>Height</th>
      <th>Weight</th>
      <th>CALC</th>
      <th>FAVC</th>
      <th>FCVC</th>
      <th>NCP</th>
      <th>SCC</th>
      <th>SMOKE</th>
      <th>CH2O</th>
      <th>f_h_w_ot</th>
      <th>FAF</th>
      <th>TUE</th>
      <th>CAEC</th>
      <th>MTRANS</th>
      <th>NObeyesdad</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>21.0</td>
      <td>Female</td>
      <td>1.62</td>
      <td>64.0</td>
      <td>no</td>
      <td>no</td>
      <td>2.0</td>
      <td>3.0</td>
      <td>no</td>
      <td>no</td>
      <td>2.0</td>
      <td>yes</td>
      <td>0.0</td>
      <td>1.0</td>
      <td>Sometimes</td>
      <td>Public_Transportation</td>
      <td>Normal_Weight</td>
    </tr>
    <tr>
      <th>1</th>
      <td>21.0</td>
      <td>Female</td>
      <td>1.52</td>
      <td>56.0</td>
      <td>Sometimes</td>
      <td>no</td>
      <td>3.0</td>
      <td>3.0</td>
      <td>yes</td>
      <td>yes</td>
      <td>3.0</td>
      <td>yes</td>
      <td>3.0</td>
      <td>0.0</td>
      <td>Sometimes</td>
      <td>Public_Transportation</td>
      <td>Normal_Weight</td>
    </tr>
    <tr>
      <th>2</th>
      <td>23.0</td>
      <td>Male</td>
      <td>1.80</td>
      <td>77.0</td>
      <td>Frequently</td>
      <td>no</td>
      <td>2.0</td>
      <td>3.0</td>
      <td>no</td>
      <td>no</td>
      <td>2.0</td>
      <td>yes</td>
      <td>2.0</td>
      <td>1.0</td>
      <td>Sometimes</td>
      <td>Public_Transportation</td>
      <td>Normal_Weight</td>
    </tr>
    <tr>
      <th>3</th>
      <td>27.0</td>
      <td>Male</td>
      <td>1.80</td>
      <td>87.0</td>
      <td>Frequently</td>
      <td>no</td>
      <td>3.0</td>
      <td>3.0</td>
      <td>no</td>
      <td>no</td>
      <td>2.0</td>
      <td>no</td>
      <td>2.0</td>
      <td>0.0</td>
      <td>Sometimes</td>
      <td>Walking</td>
      <td>Overweight_Level_I</td>
    </tr>
    <tr>
      <th>4</th>
      <td>22.0</td>
      <td>Male</td>
      <td>1.78</td>
      <td>89.8</td>
      <td>Sometimes</td>
      <td>no</td>
      <td>2.0</td>
      <td>1.0</td>
      <td>no</td>
      <td>no</td>
      <td>2.0</td>
      <td>no</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>Sometimes</td>
      <td>Public_Transportation</td>
      <td>Overweight_Level_II</td>
    </tr>
  </tbody>
</table>
</div>

- $\color{orange}{\text{Age}}$: Cuantitativa discreta.

- $\color{orange}{\text{Height}}$: Cuantitativa continua.
- $\color{orange}{\text{Gender}}$: Cualitativa nominal.
- $\color{orange}{\text{Weight}}$: Cuantitativa continua.
- $\color{orange}{\text{CALC}}$: Cualitativa ordinal.
- $\color{orange}{\text{family\_history\_with\_overweight}}$: Cualitativa nominal.
- $\color{orange}{\text{FAVC}}$: Cualitativa nominal.
- $\color{orange}{\text{FCVC}}$: Cuantitativa nominal.
- $\color{orange}{\text{NCP}}$: Cuantitativa discreta.
- $\color{orange}{\text{CAEC}}$: Cualitativa ordinal.
- $\color{orange}{\text{SMOKE}}$: Cualitativa nominal.
- $\color{orange}{\text{CH2O}}$: Cuantitativa discreta.
- $\color{orange}{\text{SCC}}$: Cualitativa nominal.
- $\color{orange}{\text{FAF}}$: Cualitativa ordinal.
- $\color{orange}{\text{TUE}}$: Cuantitativa discreta.
- $\color{orange}{\text{MTRANS}}$: Cualitativa nominal.
- $\color{orange}{\text{NObeyesdad}}$: Cualitativa ordinal.
- $\color{orange}{\text{BMI}}$: 